drafts/api-first-part-1.md

```diff
@@ -0,0 +1,312 @@
+---
+title: Effortless API-first 
+subtitle: Part one of our API-first series 🛠️
+slug: api-first-development
+tags: api, rest-api, openapi, gradle, api-first
+cover: https://cdn.hashnode.com/res/hashnode/image/upload/v1690024103909/5Im8zw6FL.jpg?auto=format
+domain: software-engineering-corner.hashnode.dev
+ignorePost: true
+hideFromHashnodeCommunity: false
+publishAs: romanutti
+---
+
+The software development world is an ever-evolving landscape. Technologies and methodologies continuously evolve - each promising a more efficient and effective way to create high-quality software. 
+One such approach that is gaining significant traction is *API-first* development.
+
+In this article series we will explore the idea behind API-first, look at practicable examples and [OpenAPI Generator](https://github.com/OpenAPITools/openapi-generator)'s new multiple specification files feature.
+
+> Note: Release 7.0.1 that will fix [issue 16419](https://github.com/OpenAPITools/openapi-generator/issues/16419) is expected to be released later this year. Until then multiple specification files are only supported by the CLI version.
+
+### What does API-first mean?
+
+API-first design is a development paradigm that prioritizes our APIs, or application programming interfaces, and how our different software pieces communicate. In this approach, APIs are designed before any line of code is written.
+
+[OpenAPI Specification](https://spec.openapis.org/oas/v3.1.0) is used to outline how the API should act. It is a standard that allows us to define endpoints, operations, parameters, error messages, and other information in a way that both humans and computers understand.
+
+A fully-fledged example of an OpenAPI specification can be found [here](https://editor.swagger.io/).
+
+### The benefits of API-first development
+
+API-first comes with a lot of benefits. Here are some of them:
+
```

Comment:
to me one of the biggest benefits is that you are actually talking about what is required by the consumer of the API.
oftentimes endpoints are provided that do not really provide what the frontend actually needs, resulting in more complexity in the frontend. with this approach, you are building a backend for frontend. and you have a common source of truth. it also facilitates the collaboration between frontend, backend and other stakeholders and ensures a sort of contract between them.